trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/signing/Signer.java

```diff
@@ -164,8 +167,17 @@ static SigningContext sign(
                 entity);
     }
 
+    private record InternalRequestAuthorization(RequestAuthorization requestAuthorization, URI signingUri)
```

Comment:
This never escapes this `Signer`. It's an internal implementation detail of `Signer`. `private` is right for this.

Reply:
Ah you are right sorry. Because of folding I though https://github.com/trinodb/aws-proxy/pull/94/files/0b30167a0018ef369cc48f41a5d62ad5978bf324#diff-015b83045b14422260caa605b1044f7b29ea6b9eb284cea9e4a4e6c9176ac0dcR100 was the top level function return